### PR TITLE
[hotfix] Align the parameter pattern of retry_times with retry_times_…

### PR DESCRIPTION
**Pull request by KarmaGYZ**
…with_backoff_and_cleanup

## What is the purpose of the change

This PR align the parameter pattern of `retry_times` and `retry_times_with_backoff_and_cleanup`.

## Brief change log

Change the parameter pattern of retry_times.
Edit the relevant e2e tests.

## Verifying this change

- Trigger e2e tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? no

cc @tillrohrmann @zhijiangW

### CODE REVIEW

**Comment by KarmaGYZ** [comment]
Travis gives green light to relevant e2e tests [here](https://travis-ci.org/KarmaGYZ/flink/builds/628975565)

**Comment by KarmaGYZ** [comment]
Kindly ping @tillrohrmann

File: flink-end-to-end-tests/test-scripts/common_mesos_docker.sh [hunk 1]
```diff
@@ -63,10 +67,9 @@ function start_flink_cluster_with_mesos() {
 
 function build_image() {
     echo "Building Mesos Docker container"
-    if ! retry_times $IMAGE_BUILD_RETRIES 0 docker build -f $END_TO_END_DIR/test-scripts/docker-mesos-cluster/Dockerfile \
+    if ! docker build -f $END_TO_END_DIR/test-scripts/docker-mesos-cluster/Dockerfile \
         -t flink/docker-mesos-cluster:latest \
         $END_TO_END_DIR/test-scripts/docker-mesos-cluster/; then
-        echo "ERROR: Could not build mesos image. Aborting..."
-        exit 1
+        return 1
```
Comment (tillrohrmann) [inline flink-end-to-end-tests/test-scripts/common_mesos_docker.sh]: Shouldn't the `docker` command alone set a non-zero return value if something went wrong? So I guess the `if` block is not really needed.

File: flink-end-to-end-tests/test-scripts/common_yarn_docker.sh [hunk 2]
```diff
@@ -91,17 +91,19 @@ function start_hadoop_cluster() {
...
 function build_image() {
     echo "Building Hadoop Docker container"
-    if ! retry_times $IMAGE_BUILD_RETRIES 2 docker build --build-arg HADOOP_VERSION=2.8.4 \
+    if ! docker build --build-arg HADOOP_VERSION=2.8.4 \
         -f $END_TO_END_DIR/test-scripts/docker-hadoop-secure-cluster/Dockerfile \
         -t flink/docker-hadoop-secure-cluster:latest \
         $END_TO_END_DIR/test-scripts/docker-hadoop-secure-cluster/; then
-        echo "ERROR: Could not build hadoop image. Aborting..."
-        exit 1
+        return 1
```
Comment (tillrohrmann) [inline flink-end-to-end-tests/test-scripts/common_yarn_docker.sh]: Same here with the if block.

**Comment by KarmaGYZ** [comment]
Thanks a lot to point out this @tillrohrmann . I update the PR.

**Pull request closed without merging**